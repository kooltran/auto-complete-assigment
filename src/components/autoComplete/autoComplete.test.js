import React from "react";
import renderer from "react-test-renderer";
import MyAutoComplete from "./autoComplete";
// import { render, fireEvent, waitFor, screen } from "@testing-library/react";

test("should render MyAutoComplete correctly", () => {
  const tree = renderer.create(<MyAutoComplete />).toJSON();
  expect(tree).toMatchSnapshot();
});

// test("render autocomplete list", () => {
//   const onSearch = jest.fn();
//   const { getByRole } = render(<MyAutoComplete />);
//   const autoCompleteInput = getByRole("combobox");
//   fireEvent.change(autoCompleteInput, { target: { value: "abc" } });
//   expect(onSearch).toHaveBeenCalledTimes(1);
// });
