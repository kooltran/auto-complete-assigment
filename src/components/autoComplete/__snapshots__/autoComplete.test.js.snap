// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render MyAutoComplete correctly 1`] = `
<div
  className="ant-select ant-select-auto-complete ant-select-single ant-select-show-search"
  data-testid="auto-complete-input"
  onBlur={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  style={
    Object {
      "width": 200,
    }
  }
>
  <div
    className="ant-select-selector"
    onClick={[Function]}
    onMouseDown={[Function]}
  >
    <span
      className="ant-select-selection-search"
    >
      <input
        aria-activedescendant="undefined_list_0"
        aria-autocomplete="list"
        aria-controls="undefined_list"
        aria-haspopup="listbox"
        aria-owns="undefined_list"
        autoComplete="off"
        className="ant-select-selection-search-input"
        data-testid="auto-complete-input"
        onChange={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onPaste={[Function]}
        readOnly={false}
        role="combobox"
        style={
          Object {
            "opacity": null,
          }
        }
        unselectable={null}
        value=""
      />
    </span>
    <span
      className="ant-select-selection-placeholder"
    >
      Try VietNam
    </span>
  </div>
</div>
`;
